File: regalaya-api/src/main/java/br/com/regalaya/address/services/service/AddressService.java
package br.com.regalaya.address.services.service;

import br.com.regalaya.address.dto.requests.CreateAddressRequest;
import br.com.regalaya.address.dto.requests.UpdateAddressRequest;
import br.com.regalaya.address.dto.responses.AddressResponse;

import java.util.List;
import java.util.UUID;

public interface AddressService {

    AddressResponse create(UUID userId, CreateAddressRequest request);

    AddressResponse update(UUID userId, UUID addressId, UpdateAddressRequest request);

    void delete(UUID userId, UUID addressId);

    AddressResponse findById(UUID userId, UUID addressId);

    List<AddressResponse> findAllByUser(UUID userId);

    AddressResponse setDefault(UUID userId, UUID addressId);
}

File: regalaya-web/src/services/address.service.ts
import { http } from '@/lib/api'
import type { Address, CreateAddressRequest, UpdateAddressRequest } from '@/types/user'

export const addressService = {
  getAll: () =>
    http.get<Address[]>('/addresses'),

  getById: (id: string) =>
    http.get<Address>(`/addresses/${id}`),

  create: (data: CreateAddressRequest) =>
    http.post<Address>('/addresses', data),

  update: (id: string, data: UpdateAddressRequest) =>
    http.put<Address>(`/addresses/${id}`, data),

  delete: (id: string) =>
    http.delete(`/addresses/${id}`),

  setDefault: (id: string) =>
    http.patch<Address>(`/addresses/${id}/default`, {}),
}
